src/library/trackset/setlogfeature.cpp

```diff
@@ -397,17 +400,23 @@ void SetlogFeature::slotPlayingTrackChanged(TrackPointer currentPlayingTrack) {
     if (m_pPlaylistTableModel->getPlaylist() == m_playlistId) {
         // View needs a refresh
 
-        WTrackTableView* view = dynamic_cast<WTrackTableView*>(
-                m_libraryWidget->getActiveView());
-        if (view != nullptr) {
-            // We have a active view on the history. The user may have some
-            // important active selection. For example putting track into crates
-            // while the song changes trough autodj. The selection is then lost
-            // and dataloss occurs
-            const QList<TrackId> trackIds = view->getSelectedTrackIds();
-            m_pPlaylistTableModel->appendTrack(currentPlayingTrackId);
-            view->setSelectedTracks(trackIds);
-        } else {
+        bool hasActiveView = false;
+        if (m_libraryWidget) {
+            WTrackTableView* view = dynamic_cast<WTrackTableView*>(
```

Comment:
Can we use qobject_cast here?

Reply:
No, I tried that but I think that caused a compiler warning.

Reply:
LibraryView is a plain C++ abstract base class.

Reply:
Super confusing because it's prefixed with a W, indicating a QWidget.